indent function needs sanity checks

### Consul Template version
```
consul-template v0.19.4 (68b1da2)
```

### Configuration

```hcl
{"consul": {"retry": {"attempts": 0, "backoff": "250ms", "enabled": true, "max_backoff": "5m"}}, "log_level": "debug"}
```

```liquid
{{ "hi" | indent -5 }}
```


### Command

```shell
consul-template -once -template 'test.tmpl:test'
```


### Debug output
```
panic: strings: negative Repeat count [recovered]
	panic: strings: negative Repeat count

goroutine 6 [running]:
text/template.errRecover(0xc42017b258)
	/usr/local/go/src/text/template/exec.go:146 +0x1c5
panic(0x83b740, 0x93fc70)
	/usr/local/go/src/runtime/panic.go:491 +0x283
strings.Repeat(0x8e7370, 0x1, 0xfffffffffffffffb, 0x4104a7, 0xc4201a6140)
	/usr/local/go/src/strings/strings.go:587 +0x1c7
github.com/hashicorp/consul-template/template.indent(0xfffffffffffffffb, 0xc420016d84, 0x2, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/hashicorp/consul-template/template/funcs.go:663 +0x55
reflect.Value.call(0x85dd60, 0x9052d0, 0x13, 0x8e77ae, 0x4, 0xc4201ac2d0, 0x2, 0x2, 0x8e6340, 0x869a01, ...)
	/usr/local/go/src/reflect/value.go:434 +0x905
reflect.Value.Call(0x85dd60, 0x9052d0, 0x13, 0xc4201ac2d0, 0x2, 0x2, 0x83b740, 0xc4201ae090, 0x98)
	/usr/local/go/src/reflect/value.go:302 +0xa4
text/template.(*state).evalCall(0xc42017b1d8, 0x0, 0x0, 0x0, 0x85dd60, 0x9052d0, 0x13, 0xb2c060, 0xc4201ac1b0, 0xc420016d8a, ...)
	/usr/local/go/src/text/template/exec.go:670 +0x580
text/template.(*state).evalFunction(0xc42017b1d8, 0x0, 0x0, 0x0, 0xc4201ac1e0, 0xb2c060, 0xc4201ac1b0, 0xc4201b6260, 0x2, 0x2, ...)
	/usr/local/go/src/text/template/exec.go:538 +0x176
text/template.(*state).evalCommand(0xc42017b1d8, 0x0, 0x0, 0x0, 0xc4201ac1b0, 0x83b740, 0xc4201ae090, 0x98, 0x83b740, 0xc4201ae090, ...)
	/usr/local/go/src/text/template/exec.go:435 +0x53a
text/template.(*state).evalPipeline(0xc42017b1d8, 0x0, 0x0, 0x0, 0xc4201b80a0, 0xc4201ac2a0, 0x30, 0x28)
	/usr/local/go/src/text/template/exec.go:408 +0x115
text/template.(*state).walk(0xc42017b1d8, 0x0, 0x0, 0x0, 0xb2bf40, 0xc4201ac240)
	/usr/local/go/src/text/template/exec.go:234 +0x4af
text/template.(*state).walk(0xc42017b1d8, 0x0, 0x0, 0x0, 0xb2c240, 0xc4201ac150)
	/usr/local/go/src/text/template/exec.go:242 +0x11d
text/template.(*Template).execute(0xc4201a6040, 0xb24960, 0xc4201f2000, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/text/template/exec.go:197 +0x1f9
text/template.(*Template).Execute(0xc4201a6040, 0xb24960, 0xc4201f2000, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/text/template/exec.go:180 +0x53
github.com/hashicorp/consul-template/template.(*Template).Execute(0xc42005ed80, 0xc420075668, 0x2, 0x2, 0xb8f240)
	/go/src/github.com/hashicorp/consul-template/template/template.go:177 +0x2d4
github.com/hashicorp/consul-template/manager.(*Runner).runTemplate(0xc4200a17a0, 0xc42005ed80, 0xc42017b8d0, 0x0, 0xc4201ac000, 0x6)
	/go/src/github.com/hashicorp/consul-template/manager/runner.go:662 +0x38c
github.com/hashicorp/consul-template/manager.(*Runner).Run(0xc4200a17a0, 0x2a, 0x0)
	/go/src/github.com/hashicorp/consul-template/manager/runner.go:512 +0x19b
github.com/hashicorp/consul-template/manager.(*Runner).Start(0xc4200a17a0)
	/go/src/github.com/hashicorp/consul-template/manager/runner.go:213 +0x12a
created by main.(*CLI).Run
	/go/src/github.com/hashicorp/consul-template/cli.go:110 +0x455

```
### Expected behavior
not a panic, but an error saying that negative numbers are not allowed


### Actual behavior
a panic and a crash


This happened when a similar template was executed as part of the template stanza in nomad and caused a crash in nomad..

--- a/template/funcs.go
+++ b/template/funcs.go
@@ -657,6 +657,9 @@ func in(l, v interface{}) (bool, error) {
 
 // Indent prefixes each line of a string with the specified number of spaces
 func indent(spaces int, s string) (string, error) {
+	if spaces < 0 {
+		return "", fmt.Errorf("indent value must be a positive integer")
+	}
 	var output, prefix []byte
 	var sp bool
 	var size int
@@ -669,7 +672,7 @@ func indent(spaces int, s string) (string, error) {
 		}
 		output = append(output, c)
 		sp = c == '\n'
-		size += 1
+		size++
 	}
 	return string(output[:size]), nil
 }

--- a/template/template_test.go
+++ b/template/template_test.go
@@ -1206,6 +1206,28 @@ func TestTemplate_Execute(t *testing.T) {
 			false,
 		},
 		{
+			"helper_indent_negative",
+			&NewTemplateInput{
+				Contents: `{{ "hello\nhello\r\nHELLO\r\nhello\nHELLO" | indent -4 }}`,
+			},
+			&ExecuteInput{
+				Brain: NewBrain(),
+			},
+			"    hello\n    hello\r\n    HELLO\r\n    hello\n    HELLO",
+			true,
+		},
+		{
+			"helper_indent_zero",
+			&NewTemplateInput{
+				Contents: `{{ "hello\nhello\r\nHELLO\r\nhello\nHELLO" | indent 0 }}`,
+			},
+			&ExecuteInput{
+				Brain: NewBrain(),
+			},
+			"hello\nhello\r\nHELLO\r\nhello\nHELLO",
+			false,
+		},
+		{
 			"helper_loop",
 			&NewTemplateInput{
 				Contents: `{{ range loop 3 }}1{{ end }}`,

